streams/src/test/java/org/apache/kafka/streams/processor/internals/ProcessorTopologyTest.java

```diff
@@ -247,6 +247,17 @@ public void testDrivingInternalRepartitioningTopology() {
     }
 
     @Test
+    public void testDrivingInternalRepartitioningTimestampTopology() {
+        driver = new ProcessorTopologyTestDriver(config, createInternalRepartitioningWithValueTimestampTopology());
+        driver.process(INPUT_TOPIC_1, "key1", "value1@1000", STRING_SERIALIZER, STRING_SERIALIZER);
+        driver.process(INPUT_TOPIC_1, "key2", "value2@2000", STRING_SERIALIZER, STRING_SERIALIZER);
+        driver.process(INPUT_TOPIC_1, "key3", "value3@3000", STRING_SERIALIZER, STRING_SERIALIZER);
+        assertNextOutputRecordTimestamp(OUTPUT_TOPIC_1, "key1", "value1", 1000L);
```

Comment:
Yep, i understand there are other `assertNextOutputRecord` methods. I'd probably replace them too (not in this PR!) . I feel using `assertThat` is better in this case as we are just comparing the `ProducerRecord`. I don't see the need to have a method with multiple assertions when it can all be done in a single assertion.